net_transport.go

```diff
@@ -80,6 +80,12 @@ type NetworkTransport struct {
 	TimeoutScale int
 }
 
+// NetworkTransportConfig encapsulates configuration for the network transport layer.
```

Comment:
One last nit - can you move the description of each thing above each member? That'll be easier to maintain and read.